azavea/raster-foundry
# Consistently use sbt-extras; move assembly builds to cipublish

## Overview

Take a tour of the various `scripts` and attempt to introduce consistency around usage of `./sbt` (sbt-extras) and `docker-compose`. Also, move certain tasks outside of `cibuild` (gets executed on every PR build) and into `cipublish` (only gets executed on `develop` builds).

- Update `sbt` to 0.13.13
- Update `sbt-extras` script
- Reroute Ivy and `sbt` boot caches into the `project` directory; get better cache reuse
- Make use of base OpenJDK container image for Scala vs. Scala container image
- Move JAR and container image builds (that aren't necessary for executing tests) into `cipublish` (only happens on `develop` builds)
- Be consistent with arguments when executing `docker-compose run`

## Testing Instructions

Execute each of the following scripts and ensure that they complete successfully:

- `bootstrap`
- `cibuild`
- `cipublish`
- `console`
- `load_development_data`
- `server`
- `setup`
- `test`

Some examples of ways to execute these commands within the virtual machine:

```bash
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/bootstrap
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/setup
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/test
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/cibuild
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/load_development_data
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/server
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/console app-frontend ls
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ AWS_ECR_ENDPOINT="615874746523.dkr.ecr.us-east-1.amazonaws.com" ./scripts/cipublish
```

It would also be good to execute `cibuild` multiple times in a row to ensure that Scala and Node dependencies aren't being pulled down multiple times.
